### PR TITLE
refactor(app): add feature flag for calibration flow redesigns 

### PR DESCRIPTION
**Pull request by b-cooper**
# Overview

Add enableCalibrationWizards feature flag and branch the orchestration components of the 4 ot2
calibration flows.

Closes RLAB-11 and RLAB-12

# Changelog

- add `enableCalibrationWizards` FF
- branch view layer using new Wizard wrapping components for orchestration components for Deck, Pipette Offset, Tip Length, and Health Check.  StepCount is stubbed for now, and will be tackled in following work for each flow.
- removes a bottom margin from the `StepMeter` component so that it's parent component can govern it's layout.
- fixes a small bug having to do with the legacy sessions not being fetched by default when the robot settings screen renders, which would keep someone from being able to pick up where they left off in a running calibration session if they navigate away or refresh the app. 
- moves `WizardHeader` component from `atoms` into `molecules` because it is composed of many atoms.
 
# Review requests

- confirm that with the feature flag set to false, nothing changes in the UX of all 4 flows
- confirm that with the feature flag set to true, each of the 4 flows renders its `CalibrationPanels` inside of the new Wizard Modal design.  

# Risk assessment
low

### CODE REVIEW

File: app/src/molecules/WizardHeader/__tests__/WizardHeader.test.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ describe('WizardHeader', () => {
   it('renders correct information with step count visible and pressing on button calls props', () => {
     const { getByText, getByRole } = render(props)
     getByText('Tip Length Calibrations')
-    const exit = getByRole('button', { name: 'exit' })
+    const exit = getByRole('button', { name: 'Exit' })
```
Comment (jerader) [inline app/src/molecules/WizardHeader/__tests__/WizardHeader.test.tsx:37]: why do we need an `aria-label` for the button?, isn't it sufficient in the test to just test for the button text?

File: app/src/molecules/WizardHeader/index.tsx [hunk 2]
```diff
@@ -45,7 +45,7 @@ export const WizardHeader = (props: WizardHeaderProps): JSX.Element => {
           ) : null}
         </Flex>
 
-        <Btn onClick={onExit}>
+        <Btn onClick={onExit} aria-label="Exit">
```
Comment (jerader) [inline app/src/molecules/WizardHeader/index.tsx:48]: why do we need this `aria-label`?

File: app/src/organisms/CalibrateDeck/__tests__/CalibrateDeck.test.tsx [hunk 3]
```diff
@@ -50,6 +56,174 @@ describe('CalibrateDeck', () => {
...
+    expect(dispatchRequests).not.toHaveBeenCalledWith(
+      Sessions.createSessionCommand('robot-name', session.id, {
+        command: Sessions.sharedCalCommands.JOG,
+        data: { vector: [0, -0.1, 0] },
+      })
+    )
+  })
+})
+
+describe('deprecated CalibrateDeck, , remove this whole block when enableCalibrationWizards FF is deleted', () => {
```
Comment (jerader) [inline app/src/organisms/CalibrateDeck/__tests__/CalibrateDeck.test.tsx:215]: interesting that you added the `remove this whole block when enableCalibrationWizards FF is deleted` in the test name and not as a comment 🤷

File: app/src/organisms/CalibratePipetteOffset/index.tsx [hunk 4]
```diff
@@ -168,7 +176,51 @@ export function CalibratePipetteOffset(
 
   // @ts-expect-error TODO protect against currentStep === undefined
   const Panel = PANEL_BY_STEP[currentStep]
-  return Panel ? (
+  if (Panel == null) return null
+  return enableCalibrationWizards ? (
+    <Portal level="top">
+      <ModalShell
+        width="47rem"
```
Comment (jerader) [inline app/src/organisms/CalibratePipetteOffset/index.tsx:183]: does this actually give us the behavior we want? I thought we wanted the modal to be center to the white background and not center to the whole viewport (ex from the pipette attach/detach flow figma)

image

File: app/src/organisms/CalibrateDeck/index.tsx [hunk 5]
```diff
@@ -154,7 +161,41 @@ export function CalibrateDeck(
   }
   // @ts-expect-error TODO: cannot index with undefined. Also, add test coverage for null case when no panel
   const Panel = PANEL_BY_STEP[currentStep]
-  return Panel ? (
+  if (Panel == null) return null
+  return enableCalibrationWizards ? (
+    <Portal level="top">
+      <ModalShell
+        width="47rem"
```
Comment (jerader) [inline app/src/organisms/CalibrateDeck/index.tsx:168]: same as other comment

File: app/src/organisms/CalibrateTipLength/index.tsx [hunk 6]
```diff
@@ -154,8 +162,40 @@ export function CalibrateTipLength(
   }
   // @ts-expect-error(sa, 2021-05-26): cannot index undefined, leaving to avoid src code change
   const Panel = PANEL_BY_STEP[currentStep]
-
-  return Panel ? (
+  if (Panel == null) return null
+  return enableCalibrationWizards ? (
+    <Portal level="top">
+      <ModalShell
+        width="47rem"
```
Comment (jerader) [inline app/src/organisms/CalibrateTipLength/index.tsx:169]: same comment

File: app/src/organisms/CalibratePipetteOffset/index.tsx [hunk 7]
```diff
@@ -168,7 +176,51 @@ export function CalibratePipetteOffset(
 
   // @ts-expect-error TODO protect against currentStep === undefined
   const Panel = PANEL_BY_STEP[currentStep]
-  return Panel ? (
+  if (Panel == null) return null
+  return enableCalibrationWizards ? (
+    <Portal level="top">
```
Comment (jerader) [inline app/src/organisms/CalibratePipetteOffset/index.tsx:181]: do any of these `Portal`s do anything?

File: app/src/organisms/Devices/hooks/useRunStartedOrLegacySessionInProgress.ts [hunk 8]
```diff
@@ -7,6 +7,7 @@ export function useRunStartedOrLegacySessionInProgress(): boolean {
   const runId = useCurrentRunId()
   const runStatus = useRunStatus(runId)
   const allSessionsQueryResponse = useAllSessionsQuery()
+  console.log(allSessionsQueryResponse.data)
```
Comment (jerader) [inline app/src/organisms/Devices/hooks/useRunStartedOrLegacySessionInProgress.ts]:
```suggestion
```

File: app/src/organisms/CalibratePipetteOffset/index.tsx (hunk 4 above)
Comment (b-cooper) [reply app/src/organisms/CalibratePipetteOffset/index.tsx:183]: Just talked with @emilywools in person, and we discussed that the desired behavior (probably in the pipette attach/detach modal too) would be to center the modal portion to the whole viewport. She's checking in with Mel to confirm.

File: app/src/organisms/CalibrateDeck/__tests__/CalibrateDeck.test.tsx (hunk 3 above)
Comment (b-cooper) [reply app/src/organisms/CalibrateDeck/__tests__/CalibrateDeck.test.tsx:215]: yeah, either works, I added it in the test name so that it appears in CI or whenever the test is running too.  It wouldn't be a bad idea to add a TODO comment here as well

File: app/src/molecules/WizardHeader/index.tsx (hunk 2 above)
Comment (b-cooper) [reply app/src/molecules/WizardHeader/index.tsx:48]: For this PR, it's added so that our old enzyme based tests can easily access this button in the test environment. In general though, it is an accessibility concern that is more necessary for buttons with symbols or icons as their contents e.g. "X" instead of exit (see [WCAG](https://www.w3.org/TR/WCAG20-TECHS/ARIA14.html)).  It's still helpful here, because we currently have the `<button>` tag wrapping a `<p>` tag, so it's ensures that the button's meaning is inferable by screen readers.

File: app/src/organisms/CalibratePipetteOffset/index.tsx (hunk 7 above)
Comment (b-cooper) [reply app/src/organisms/CalibratePipetteOffset/index.tsx:181]: These `Portals` allow us to render the modal outside of the current parent child hierarchy. This is useful because we want the overlay to span the entire viewport of the app.

File: app/src/organisms/CalibrateDeck/index.tsx [hunk 9]
```diff
@@ -149,12 +159,53 @@ export function CalibrateDeck(
...
+            </Flex>
+          ) : (
+            <Panel
+              sendCommands={sendCommands}
+              cleanUpAndExit={cleanUpAndExit}
+              tipRack={tipRack}
+              isMulti={isMulti}
+              mount={instrument?.mount.toLowerCase() as Mount}
+              currentStep={currentStep}
+              sessionType={session.sessionType}
```
Comment (koji) [inline app/src/organisms/CalibrateDeck/index.tsx]: Object is possibly 'null'.ts(2531)

```suggestion
              sessionType={session?.sessionType}
```

File: app/src/organisms/CalibratePipetteOffset/index.tsx [hunk 10]
```diff
@@ -4,19 +4,22 @@ import * as React from 'react'
...
   ModalPage,
+  Box,
   SpinnerModalPage,
   useConditionalConfirm,
   DISPLAY_FLEX,
   DIRECTION_COLUMN,
   ALIGN_CENTER,
   JUSTIFY_CENTER,
+  SPACING,
   SPACING_3,
```
Comment (koji) [inline app/src/organisms/CalibratePipetteOffset/index.tsx:15]: It would be good if we removed the old const for spacing and colors.

Comment (b-cooper) [reply app/src/organisms/CalibratePipetteOffset/index.tsx:15]: That's a good callout, it's only still there because I left the old code "not behind the FF" untouched in this file, and it will be deleted when that feature flag is ready to be deleted.

File: app/src/organisms/CalibrateDeck/index.tsx (hunk 9 above)
Comment (b-cooper) [reply app/src/organisms/CalibrateDeck/index.tsx]: There is an early exit above this point in the component that protects from this object being null.  

```
 if (!session || !tipRack) {
    return null
  }
```

File: app/src/organisms/CalibrateTipLength/index.tsx [hunk 11]
```diff
@@ -4,19 +4,22 @@ import * as React from 'react'
...
   ModalPage,
+  Box,
   SpinnerModalPage,
   useConditionalConfirm,
   DISPLAY_FLEX,
   DIRECTION_COLUMN,
   ALIGN_CENTER,
   JUSTIFY_CENTER,
+  SPACING,
   SPACING_3,
```
Comment (koji) [inline app/src/organisms/CalibrateTipLength/index.tsx:15]: It would be good if we removed the old const for spacing and colors.

**Review by brenthagen — APPROVED** [review]
changes make sense to me ✨

File: app/src/organisms/Devices/ProtocolRun/__tests__/SetupTipLengthCalibrationButton.test.tsx [hunk 12]
```diff
@@ -10,7 +10,10 @@ import fixture_tiprack_300_ul from '@opentrons/shared-data/labware/fixtures/2/fi
 import { i18n } from '../../../../i18n'
 import { AskForCalibrationBlockModal } from '../../../../organisms/CalibrateTipLength/AskForCalibrationBlockModal'
 import { mockDeckCalData } from '../../../../redux/calibration/__fixtures__'
-import { getHasCalibrationBlock } from '../../../../redux/config'
+import {
+  getHasCalibrationBlock,
+  useFeatureFlag,
+} from '../../../../redux/config'
```
Comment (brenthagen) [inline app/src/organisms/Devices/ProtocolRun/__tests__/SetupTipLengthCalibrationButton.test.tsx:16]: hooks file isn't being mocked - change line 27 to `jest.mock('../../../../redux/config')`

File: app/src/molecules/WizardHeader/__tests__/WizardHeader.test.tsx [hunk 13]
```diff
@@ -2,10 +2,10 @@ import * as React from 'react'
 import { fireEvent, screen } from '@testing-library/react'
 import { renderWithProviders } from '@opentrons/components'
 import { i18n } from '../../../i18n'
-import { StepMeter } from '../../StepMeter'
+import { StepMeter } from '../../../atoms/StepMeter'
 import { WizardHeader } from '..'
 
-jest.mock('../../StepMeter')
+jest.mock('../../../StepMeter')
```
Comment (brenthagen) [inline app/src/molecules/WizardHeader/__tests__/WizardHeader.test.tsx]: `jest.mock('../../../atoms/StepMeter')`

**Review by jerader — APPROVED** [review]
✨
